pkg/util/controller_finder.go

```diff
@@ -94,6 +94,13 @@ func (r *ControllerFinder) GetWorkloadForRef(rollout *rolloutv1beta1.Rollout) (*
 				return workload, err
 			}
 		}
+	} else if rollout.Spec.Strategy.GetRollingStyle() == rolloutv1beta1.BlueGreenRollingStyle {
+		for _, finder := range r.bluegreenStyleFinders() {
+			workload, err := finder(rollout.Namespace, &workloadRef)
+			if workload != nil || err != nil {
+				return workload, err
+			}
+		}
```

Comment:
consider extract the following code block  as a common code

```
		for _, finder := range finders {
			workload, err := finder(rollout.Namespace, &workloadRef)
			if workload != nil || err != nil {
				return workload, err
			}
		}
```

Reply:
ok